pkg/appsec/appsec.go

```diff
@@ -0,0 +1,89 @@
+package appsec
+
+import (
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"time"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/pkg/errors"
+)
+
+// ErrAgentDisabled is the error message logged when the AppSec agent is
+// disabled by configuration.
+var ErrAgentDisabled = errors.New("AppSec agent disabled. Set the " +
+	"environment variable `DD_APPSEC_ENABLED=true` or add the entry " +
+	"`appsec_config.enabled: true` to your datadog.yaml file")
+
+// NewIntakeReverseProxy returns the AppSec Intake Proxy handler according to
+// the agent configuration.
+func NewIntakeReverseProxy(transport *http.Transport) (http.Handler, error) {
+	disabled := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "appsec api disabled", http.StatusNotImplemented)
+	})
+	cfg, err := newConfig(coreconfig.Datadog)
+	if err != nil {
+		return disabled, errors.Wrap(err, "configuration: ")
+	}
+	if !cfg.Enabled {
+		return disabled, ErrAgentDisabled
```

Comment:
I don't think this qualifies as an error messages. Disabling appsec is a potentially valid user intention.
```suggestion
		return disabled, nil
```

Reply:
Ok, but I also added a log message before returning.